File: back-end/src/main/java/com/meutcc/backend/content/lesson/LessonDTO.java
package com.meutcc.backend.content.lesson;

import com.meutcc.backend.content.attachment.AttachmentDTO;
import com.meutcc.backend.content.exercise.ExerciseResponseDTO;
import com.meutcc.backend.content.video.VideoDTO;
import jakarta.validation.constraints.Size;

import java.util.List;

public record LessonDTO(
        Long id,
        @Size(min = 1, max = 100)
        String title,
        @Size(min = 1, max = 1000)
        String description,
        Integer orderIndex,
        Integer durationMinutes,
        Long moduleId,
        String moduleName,
        List<VideoDTO> videos,
        List<AttachmentDTO> attachments,
        List<ExerciseResponseDTO> exercises
) {

}

File: front-end/src/types/LessonDTO.ts
import type { AttachmentDTO } from "./AttachmentDTO";
import type { ExerciseResponseDTO } from "./ExerciseResponseDTO";

export interface LessonDTO {
  id?: number;
  title: string;
  description: string;
  orderIndex: number;
  durationMinutes?: number;
  moduleId: number;
  videos?: VideoDTO[];
  attachments?: AttachmentDTO[];
  exercises?: ExerciseResponseDTO[]; 
}
export interface VideoDTO {
  duration: number;
  id?: number;
  title: string;
  url?: string;
  storageType: "URL" | "DATABASE" | "FILE_SYSTEM";
  lessonId?: number;
}
